dokka-subprojects/analysis-kotlin-api/src/test/kotlin/org/jetbrains/dokka/analysis/test/jvm/java/JavadocAnalysisTest.kt

```diff
@@ -0,0 +1,122 @@
+                     * @param <Bar> type parameter,
+                     *  long type parameter description
+                     */
+                    public class Foo<Bar> {}
+                """
+            }
+        }
+
+        val module = testProject.parse()
+        val pkg = module.packages.single()
+        val cls = pkg.classlikes.single()
+
+        assertDocumentationNodeContains(
+            cls.documentation,
+            listOf(
+                Description(customDocTag("class")),
+                Param(customDocTag("type parameter, long type parameter description"), "<Bar>"),
+            )
+        )
+    }
+
+    @Test
+    fun `should parse javadoc @param for type parameter on function`() {
+        val testProject = javaTestProject {
+            javaFile(pathFromSrc = "Foo.java") {
+                +"""
+                    public class Foo {
+                        /**
+                         * function
+                         *
+                         * @param <Bar> type parameter,
+                         *  long type parameter description
+                         */
+                        public <Bar> void something(Bar bar) {}
+                    }
+                """
+            }
+        }
+
+        val module = testProject.parse()
+        val pkg = module.packages.single()
+        val cls = pkg.classlikes.single()
+        val function = cls.functions.single { it.name == "something" }
+
+        assertDocumentationNodeContains(
+            function.documentation,
+            listOf(
+                Description(customDocTag("function")),
+                Param(customDocTag("type parameter, long type parameter description"), "<Bar>"),
+            )
+        )
+    }
+
+    @Test
+    fun `should parse javadoc @param for parameter on function`() {
+        val testProject = javaTestProject {
+            javaFile(pathFromSrc = "Foo.java") {
+                +"""
+                    public class Foo {
+                        /**
+                         * function
+                         *
+                         * @param bar parameter,
+                         *  long parameter description
+                         */
+                        public void something(String bar) {}
+                    }
+                """
+            }
+        }
+
+        val module = testProject.parse()
+        val pkg = module.packages.single()
+        val cls = pkg.classlikes.single()
+        val function = cls.functions.single { it.name == "something" }
+
+        assertDocumentationNodeContains(
+            function.documentation,
+            listOf(
+                Description(customDocTag("function")),
+                Param(customDocTag("parameter, long parameter description"), "bar"),
+            )
+        )
+    }
+
+    private fun customDocTag(text: String): CustomDocTag {
+        return CustomDocTag(listOf(P(listOf(Text(text)))), name = "MARKDOWN_FILE")
```

Comment:
So, `MARKDOWN_FILE` is needed here just because it's already created in other code — here it's needed only to test that the content is the same.

But overall, I've tracked why this was introduced and will write some info into #3366